nuxeo-compound-documents/src/test/java/org/nuxeo/compound/documents/TestCompoundDocumentImporter.java

```diff
@@ -63,4 +70,61 @@ public void testCreateCompoundDocument() throws IOException {
         assertCompoundDocument(doc);
     }
 
+    @Test
+    public void testBadArchive() throws IOException {
+        Blob blob = getBadArchive();
+        FileImporterContext context = FileImporterContext.builder(session, blob, "/").build();
+
+        NuxeoException e = assertThrows(NuxeoException.class, () -> fileManager.createOrUpdateDocument(context));
+
+        String compoundDocName = FilenameUtils.removeExtension(blob.getFilename());
+        assertEquals(String.format("Failed to create CompoundDocument for archive: %s.zip in parent: %s",
+                compoundDocName, context.getParentPath()), e.getMessage());
```

Comment:
```suggestion
        assertEquals(String.format("Failed to create CompoundDocument for archive: %s in parent: %s",
                blob.getFilename(), context.getParentPath()), e.getMessage());
```
?

Didn't see but the Blobs returned by `getTestArchive` and `getBadArchive` should have a filename ending with `.zip` - like in the real world.

Reply:
Here is the name of a generated archive. (the numbers correspond to the temp aspect of the file) ```/Users/Nour.Alkotob/git/nuxeo-compound-documents/nuxeo-compound-documents/target/nxruntime-main-18159674292140971573.tmp/tmp/test2128821898820322689.zip```
It is set there: https://github.com/nuxeo/nuxeo-compound-documents/pull/8/files#diff-1535a0fec3b983de25be45a8ed2f129aea7e55e51fb835893c1896d8f2313fccR50
and there https://github.com/nuxeo/nuxeo-compound-documents/pull/8/files#diff-1535a0fec3b983de25be45a8ed2f129aea7e55e51fb835893c1896d8f2313fccR56
I think this is why the FileManager treats it as a zip at import.

Reply:
Agree with @troger, the blob filename should be `test.zip` and not the file name.